pyoozie/tags.py

```diff
@@ -504,26 +505,25 @@ class _AbstractWorkflowEntity(typing.Iterable):
 
     def __init__(
             self,
-            xml_tag,       # typing.Text
+            xml_tag=None,  # type: typing.Optional[typing.Text]
             name=None,     # type: typing.Optional[typing.Text]
             on_error=None  # type: typing.Optional[_AbstractWorkflowEntity]
     ):
         # type: (...) -> None
         self.__xml_tag = xml_tag
+        self.__name = name if name else uuid.uuid4().hex[:8]
         self.__on_error = copy.deepcopy(on_error)
-        if name:
-            self.__identifier = '{tag}-{name}'.format(tag=xml_tag, name=name)
-        else:
-            self.__identifier = '{tag}-{uid}'.format(tag=xml_tag, uid=uuid.uuid4().hex[:8])
-
-    def identifier(self):
-        # type: () -> typing.Text
-        return self.__identifier
 
     def xml_tag(self):
         # type: () -> typing.Text
         return self.__xml_tag
 
+    def identifier(self, xml_tag=None):
```

Comment:
It is doing two different things... Refactored into two methods. `identifier()` is not used to get the identifier for an entity so that you can transition to it and and `create_identifier` can be used to create a similar identifier (used in `Parallel`).